customers.js

```diff
@@ -78,14 +80,17 @@ addCustomerToServiceDesk = async (account) => {
 
 //returns account ID of customer
 exports.getCustomerRecord = async (email, name) => {
+  // TODO: make sure we have something that looks like a complete email address?
+  // TODO: What if the email address matches multiple things?
```

Comment:
It can't? I don't think Jira will let you have the same email for multiple user accounts. It's the reason I'm using an email address instead of a user display name for the search query parameter. If I search for 'caryl' I get like 7 dummy accounts, but they all have different (fake) email addresses. 

If a user typos their email address, the addCustomer function will make them a new customer account. 

This is all a workaround in order to get the user email address in the "raised on behalf of" Jira field, and it _works_, but it's not ideal.

Reply:
It might be a bit of a pathological case -- the documentation (https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-user-search/#api-rest-api-3-user-search-get) says:

> A query string that is matched against user attributes ( displayName, and emailAddress) to find relevant users. The string can match the prefix of the attribute's value. For example, query=john matches a user with a displayName of John Smith and a user with an emailAddress of [johnson@example.com](mailto:johnson@example.com). Required, unless accountId or property is specified.

What happens if there are two users where one has an email address that's a prefix of another? I have a hard time thinking of a case where that could legitimately happen, and one would hope that it's already the case that jira will return the best match first -- so this is probably fine as-is, so long as we do some enforcement of ensuring there's something that looks like there's a complete email address.

Reply:
I guess I didn't think of a situation in which something like `admin@umich.edu` would also match `hathitrust-admin@umich.edu`. Good catch!